src/net.cpp

```diff
@@ -1388,114 +1388,189 @@ void CConnman::WakeMessageHandler()
+        // If none of those are available, we fallback on to fixed seeds immediately, else we allow
+        // 60 seconds for any of those sources to populate addrman.
+        bool add_fixed_seeds_now = false;
+        // It is cheapest to check if enough time has passed first.
+        if (GetTime<std::chrono::seconds>() > start + std::chrono::minutes{1}) {
+            add_fixed_seeds_now = true;
+            LogPrintf("Adding fixed seeds as 60 seconds have passed and addrman is empty\n");
+        }
+
+        // Checking !dnsseed is cheaper before locking 2 mutexes.
+        if (!add_fixed_seeds_now && !dnsseed) {
+            LOCK2(m_addr_fetches_mutex, m_added_nodes_mutex);
+            if (m_addr_fetches.empty() && m_added_nodes.empty()) {
+                add_fixed_seeds_now = true;
+                LogPrintf("Adding fixed seeds as -dnsseed=0 (or IPv4/IPv6 connections are disabled via -onlynet), -addnode is not provided and all -seednode(s) attempted\n");
             }
-            unsigned int nMaxIPs = 256; // Limits number of IPs learned from a DNS seed
-            if (LookupHost(host, vIPs, nMaxIPs, true)) {
-                for (const CNetAddr& ip : vIPs) {
-                    CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()), requiredServiceBits);
-                    addr.nTime = rng.rand_uniform_delay(Now<NodeSeconds>() - 3 * 24h, -4 * 24h); // use a random age between 3 and 7 days old
-                    vAdd.push_back(addr);
-                    found++;
+        }
+
+        if (add_fixed_seeds_now) {
+            std::vector<CAddress> seed_addrs{ConvertSeeds(Params().FixedSeeds())};
+            // We will not make outgoing connections to peers that are unreachable
+            // (e.g. because of -onlynet configuration).
+            // Therefore, we do not add them to addrman in the first place.
+            // Note that if you change -onlynet setting from one network to another,
+            // peers.dat will contain only peers of unreachable networks and
+            // manual intervention will be needed (either delete peers.dat after
+            // configuration change or manually add some reachable peer using addnode),
+            // see <https://github.com/bitcoin/bitcoin/issues/26035> for details.
+            seed_addrs.erase(std::remove_if(seed_addrs.begin(), seed_addrs.end(),
+                                            [](const CAddress& addr) { return !IsReachable(addr); }),
+                             seed_addrs.end());
+            Shuffle(seed_addrs.begin(), seed_addrs.end(), FastRandomContext());
+
+            // Make AddrFetch connections to fixed seeds first. This reduces the
+            // load on the fixed seeds that would otherwise be serving blocks for
+            // many new peers requiring IBD.
+            LogPrintf("Initiating AddrFetch connections to fixed seeds.\n");
+            for (size_t addr_pos = 0; addr_pos < 10; ++addr_pos) {
```

Comment:
e4388787870ac2b42903cde47152e4735eb1be8 it might be helpful to explain/document the magic value of 10, similar to the commit message

Reply:
Done.